### PR TITLE
Dev - Upgrade to Node v20

### LINKED ISSUE #9116: Upgrade node support to version 20
### Description
Currently, the supported node version is set to `v16` in both `.nvmrc` and `package.json`. According to the current release schedule https://nodejs.org/en/about/previous-releases, v16 is out of maintenance mode, with v20 being the active LTS version.

Since WooCommerce Core already performed the upgrade to v20 a few months ago https://github.com/woocommerce/woocommerce/pull/45148 we should do the same for WooPayments.

### Acceptance criteria

* All the functionality must remain the same.
* All tests must pass.
* All scripts defined in `package.json` should still work as expected.

### Testing instructions

- All tests, checks, and scripts should pass after the upgrade
- Functionality should be unchanged.

### Additional context
* This issue currently blocks #8581 
* The GH actions will be upgraded as part of a separate issue: #8129

### PR DESCRIPTION
**Pull request by danielmx-dev**
Fixes #9116

#### Changes proposed in this Pull Request

- Upgrade Node version to v20
- Upgrade packages that require a specific version of node. Most notably: `@woocommerce/components` and `@wordpress/scripts`. Other packages related to testing and linting were also updated.
- Update test snapshots
- Replace deprecated functions in tests
- Perform various fixes in tests so they work properly with the new versions of the packages that were upgraded.
- E2E Tests are being fixed in this other PR https://github.com/Automattic/woocommerce-payments/pull/9132

#### Testing instructions

* Make sure npm scripts work as expected locally: `npm build`, `npm start`, `npm lint`, `npm test:js`
* Make sure all checks are passing in the PR: css, lint, js, php
* You can also test the build in a Jurassic Ninja site; however, since no client code was changed, we don't expect any changes in build size and functionality.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9121` or branch name `dev/upgrade-support-to-node-20` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=dev/upgrade-support-to-node-20) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 4579ce1844dd5d0bd50b03a86d8f5d7fc6505c6f
- Build time: 2024-07-23 16:13:31 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by danielmx-dev** [comment]
The `Compressed Size` check is currently failing due to node version upgrade (since the action doesn't change node versions between builds). We have a couple options:
1. Update `.npmrc` options to `engine-strict=false`, merge to develop, then update this PR so it can perform the comparison.
2. Move forward with the failing check. Since in the other PR I have confirmed that the action works as expected when both the base and target branches are using the same node version. See: https://github.com/Automattic/woocommerce-payments/pull/9132#issuecomment-2237508894

File: client/components/tooltip/test/index.test.tsx [hunk 1]
```diff
@@ -193,13 +193,17 @@ describe( 'ClickTooltip', () => {
 
 		act( () => {
 			userEvent.click( screen.getByText( 'Trigger element' ) );
+		} );
+
+		expect( handleHideMock ).toHaveBeenCalled();
+
+		act( () => {
```
Comment (frosso) [inline client/components/tooltip/test/index.test.tsx]: I'm curious about this change - why did the expect get moved from the end to the middle here? Just to be closer to the `click` action?

File: client/payment-methods/capability-request/types.ts [hunk 2]
```diff
@@ -1,7 +1,5 @@
-/**
- * External dependencies
- */
-import { Status } from '@wordpress/notices';
+// This type was imported from @wordpress/notices but it's not longer exported by the module.
```
Comment (frosso) [inline client/payment-methods/capability-request/types.ts:1]: Thank you for adding the inline comment 👍

**Review by frosso — APPROVED** [review]
Awesome job!! 💯  Those snapshot updates are 🤢 I think we should move away from using snapshots.

Looks like an additional update is needed on this action - we should be able to just use the latest version: https://github.com/Automattic/woocommerce-payments/blob/bd5fa2793ec195315d3a2f21bc0157c413d6a8ba/.github/workflows/bundle-size.yml

Tested locally in settings/disputes/checkout/etc. I didn't notice any differences.

Thank you for working on this!

File: client/components/tooltip/test/index.test.tsx (hunk 1 above)
Comment (danielmx-dev) [reply client/components/tooltip/test/index.test.tsx]: You are right, it wasn't necessary, the only necessary change was to wrap the runAllTimers call in a separate `act()` block. I'll revert that small change.

**Pull request merged**